File: xprdata/public/js/admin_dashboard.js
(function ($, api_url, Vue, swal, axios, location, moment, undefined) {
    $.fn.ddatepicker = $.fn.datepicker; // jquery-ui is overriding the bootstrap-datepicker

    const client = commissionEngine.createAccessClient('admin/dashboard');
    commissionEngine.setupAccessTokenJQueryAjax();

    let $dt = null;
    let $new_members = null;
    let $pack_sales = null;
    let $endorsers = null;

    const vm = new Vue({
        el: '#dashboard',
        data: {
            new_customer_count: null,
            new_customer_with_product_subscription_count: null,
            new_endorser_count: null,
            new_endorser_with_product_subscription_count: null,
            customer_transformation_pack_total_sales: null,
            transformation_pack_total_sales: null,
            elite_pack_total_sales: null,
            family_elite_pack_total_sales: null,
            average_reorder: null,
            viral_index: null,
            viral_index_start_date: null,
            viral_index_end_date: null,
            error: null,
            start_date: null,
            end_date: null,
            is_downloading: false,
            report_type: "",
            is_downloading_top_endorser: false,
            top_endorser_count: 0,
            user_id: 0,
            pack_count: 0,
        },
        mounted() {
            $dt = $("#table-top-endorsers").DataTable({
                searching: false,
                ordering: false,
                lengthChange: false,
                columns: [
                    {data: 'user_id', className: "text-center"},
                    {data: 'endorser'},
                    {
                        data: 'endorser_count',
                        className: "text-center",
                        render: function ( data, type, row, meta ) {
                            return `<a class="btn-link btn-view-endorsers">${data}</a>`;
                        }
                    },
                    {data: 'volume', className: "text-right"},
                ]
            });

            $("#table-top-endorsers").on('click', '.btn-view-endorsers', function () {
                let data = $dt.row($(this).parents('tr')).data();
                vm.viewEndorsers(+data.user_id);
            });

            $new_members = $("#table-new-members").DataTable({
                responsive: true,
                columns: [
                    {data: 'member_id', className: "text-center"},
                    {data: 'member'},
                    {data: 'sponsor_id', className: "text-center"},
                    {data: 'sponsor'},
                    {data: 'sponsor_type'},
                    {data: 'order_id'},
                    {data: 'cv'},
                    {data: 'amount_paid'},
                    {data: 'has_coupon'},
                    {data: 'has_gift_card'},
                    {data: 'shipping_city'},
                    {data: 'shipping_state'},
                    {data: 'has_subscription'},
                    {data: 'cellphone'},

                ]
            });

            $pack_sales = $("#table-pack-sales").DataTable({
                responsive: true,
                columns: [
                    {data: 'purchaser_id', className: "text-center"},
                    {data: 'purchaser'},
                    {data: 'invoice', className: "text-center"},
                    {data: 'sponsor_id', className: "text-center"},
                    {data: 'sponsor'},
                    {data: 'order_id'},
                    {data: 'cv'},
                    {data: 'amount_paid'},
                    {data: 'has_coupon'},
                    {data: 'has_gift_card'},
                    {data: 'shipping_city'},
                    {data: 'shipping_state'},
                    {data: 'sponsor_type'},
                ]
            });

            $endorsers = $("#table-modal-view-endorsers").DataTable({
                responsive: true,
                columns: [
                    {data: 'member_id', className: "text-center"},
                    {data: 'member'},
                    {data: 'invoice', className: "text-center"},
                    {data: 'description'},
                    {data: 'transaction_date'},
                    {data: 'sponsor_id', className: "text-center"},
                    {data: 'sponsor'},
                    {data: 'order_id'},
                    {data: 'cv'},
                    {data: 'amount_paid'},
                    {data: 'has_coupon'},
                    {data: 'has_gift_card'},
                    {data: 'shipping_city'},
                    {data: 'shipping_state'},
                    {data: 'sponsor_type'},
                    {data: 'cellphone'},
                ]
            });

            $('#start-date').ddatepicker({
                "setDate" : new Date(),
                "format": "yyyy-mm-dd"
            }).on('changeDate', function(e){
                $('#end-date').ddatepicker('setStartDate' , e.date);

                if($('#end-date').ddatepicker('getDate') < e.date) {
                    $('#end-date').ddatepicker('setDate', e.date);
                }
            });

            $('#end-date').ddatepicker({
                "setDate" : new Date(),
                "startDate" : new Date(),
                "format": "yyyy-mm-dd"
            });

            $('#start-date').ddatepicker('setDate', new Date());
            $('#end-date').ddatepicker('setDate', new Date());

            this.view();
        },
        methods: {
            getNewCustomerCount() {
                this.new_customer_count = null;
                client.get(`new-customer-count?start_date=${this.start_date}&end_date=${this.end_date}`).then(response => {
                    this.new_customer_count = +response.data.count;
                }).catch(this.axiosErrorHandler);
            },
            getNewCustomerWithProductSubscriptionCount() {
                this.new_customer_with_product_subscription_count = null;
                client.get(`new-customer-with-product-subscription-count?start_date=${this.start_date}&end_date=${this.end_date}`).then(response => {
                    this.new_customer_with_product_subscription_count = +response.data.count;
                }).catch(this.axiosErrorHandler);
            },
            getNewEndorserCount() {
                this.new_endorser_count = null;
                client.get(`new-endorser-count?start_date=${this.start_date}&end_date=${this.end_date}`).then(response => {
                    this.new_endorser_count = +response.data.count;
                }).catch(this.axiosErrorHandler);
            },
            getNewEndorserWithProductSubscriptionCount() {
                this.new_endorser_with_product_subscription_count = null;
                client.get(`new-endorser-with-product-subscription-count?start_date=${this.start_date}&end_date=${this.end_date}`).then(response => {
                    this.new_endorser_with_product_subscription_count = +response.data.count;
                }).catch(this.axiosErrorHandler);
            },
            getCustomerTransformationPackTotalSales() {
                this.customer_transformation_pack_total_sales = null;
                client.get(`total-sales/customer-transformation-pack?start_date=${this.start_date}&end_date=${this.end_date}`).then(response => {
                    this.customer_transformation_pack_total_sales = +response.data.total_sales;
                }).catch(this.axiosErrorHandler);
            },
            getTransformationPackTotalSales() {
                this.transformation_pack_total_sales = null;
                client.get(`total-sales/transformation-pack?start_date=${this.start_date}&end_date=${this.end_date}`).then(response => {
                    this.transformation_pack_total_sales = +response.data.total_sales;
                }).catch(this.axiosErrorHandler);
            },
            getElitePackTotalSales() {
                this.elite_pack_total_sales = null;
                client.get(`total-sales/elite-pack?start_date=${this.start_date}&end_date=${this.end_date}`).then(response => {
                    this.elite_pack_total_sales = +response.data.total_sales;
                }).catch(this.axiosErrorHandler);
            },
            getFamilyElitePackTotalSales() {
                this.family_elite_pack_total_sales = null;
                client.get(`total-sales/family-elite-pack?start_date=${this.start_date}&end_date=${this.end_date}`).then(response => {
                    this.family_elite_pack_total_sales = +response.data.total_sales;
                }).catch(this.axiosErrorHandler);
            },
            getAveragaReorder() {
                this.average_reorder = null;
                client.get(`average-reorder?start_date=${this.start_date}&end_date=${this.end_date}`).then(response => {
                    this.average_reorder = +response.data.average;
                }).catch(this.axiosErrorHandler);
            },
            getTopEndorsers() {
                $dt.clear().draw();
                this.top_endorser_count = 0;
                client.get(`top-endorsers?start_date=${this.start_date}&end_date=${this.end_date}`).then(response => {
                    this.top_endorser_count = response.data.length;
                    $dt.rows.add(response.data);
                    $dt.columns.adjust().draw();
                }).catch(this.axiosErrorHandler);
            },
            getViralIndex() {
                this.viral_index = null;
                this.viral_index_start_date = null;
                this.viral_index_end_date = null;

                client.get(`viral-index?start_date=${this.start_date}&end_date=${this.end_date}`).then(response => {
                    let data = response.data;
                    this.viral_index = data.viral_index;
                    this.viral_index_start_date = data.this_month.start_date;
                    this.viral_index_end_date = data.this_month.end_date;
                });
            },
            view() {
                this.start_date = $('#start-date').val();
                this.end_date = $('#end-date').val();

                this.getNewCustomerCount();
                this.getNewCustomerWithProductSubscriptionCount();
                this.getNewEndorserCount();
                this.getNewEndorserWithProductSubscriptionCount();
                this.getCustomerTransformationPackTotalSales();
                this.getTransformationPackTotalSales();
                this.getElitePackTotalSales();
                this.getFamilyElitePackTotalSales();
                this.getAveragaReorder();
                this.getTopEndorsers();
                this.getViralIndex();
            },
            viewNewCustomers() {
                this.report_type = "NEW_CUSTOMERS";
                $new_members.clear().draw();
                $('#modal-view-new-members-title').text("New Customers");
                $('#modal-view-new-members').modal({backdrop: 'static', keyboard: false});
                $new_members.responsive.recalc();

                client.get(`new-customers?start_date=${this.start_date}&end_date=${this.end_date}`).then(response => {
                    $new_members.rows.add(response.data);
                    $new_members.columns.adjust().draw();
                    $new_members.responsive.recalc();
                });

            },
            viewNewCustomersWithProductSubscription() {
                this.report_type = "NEW_CUSTOMERS_PS";
                $new_members.clear().draw();
                $('#modal-view-new-members-title').text("New Customers with Product Subscription");
                $('#modal-view-new-members').modal({backdrop: 'static', keyboard: false});
                $new_members.responsive.recalc();

                client.get(`new-customers-with-product-subscription?start_date=${this.start_date}&end_date=${this.end_date}`).then(response => {
                    $new_members.rows.add(response.data);
                    $new_members.columns.adjust().draw();
                    $new_members.responsive.recalc();
                });

            },
            viewNewEndorsers() {
                this.report_type = "NEW_ENDORSERS";
                $new_members.clear().draw();
                $('#modal-view-new-members-title').text("New Endorsers");
                $('#modal-view-new-members').modal({backdrop: 'static', keyboard: false});
                $new_members.responsive.recalc();

                client.get(`new-endorsers?start_date=${this.start_date}&end_date=${this.end_date}`).then(response => {
                    $new_members.rows.add(response.data);
                    $new_members.columns.adjust().draw();
                    $new_members.responsive.recalc();
                });
            },
            viewNewEndorsersWithProductSubscription() {
                this.report_type = "NEW_ENDORSERS_PS";
                $new_members.clear().draw();
                $('#modal-view-new-members-title').text("New Endorsers with Product Subscription");
                $('#modal-view-new-members').modal({backdrop: 'static', keyboard: false});
                $new_members.responsive.recalc();

                client.get(`new-endorsers-with-product-subscription?start_date=${this.start_date}&end_date=${this.end_date}`).then(response => {
                    $new_members.rows.add(response.data);
                    $new_members.columns.adjust().draw();
                    $new_members.responsive.recalc();
                });
            },
            viewCustomerTransformationPackSales() {
                this.report_type = "CUSTOMER_TRANSFORMATION_PACK";
                $pack_sales.clear().draw();
                $('#modal-view-pack-sales-title').text("Customer Transformation Pack Sales");
                $('#modal-view-pack-sales').modal({backdrop: 'static', keyboard: false});

                client.get(`sales/customer-transformation-pack?start_date=${this.start_date}&end_date=${this.end_date}`).then(response => {
                    $pack_sales.rows.add(response.data);
                    $pack_sales.columns.adjust().draw();
                    $pack_sales.responsive.recalc();
                });
            },
            viewTransformationPackSales() {
                this.report_type = "TRANSFORMATION_PACK";
                $pack_sales.clear().draw();
                $('#modal-view-pack-sales-title').text("Transformation Pack Sales");
                $('#modal-view-pack-sales').modal({backdrop: 'static', keyboard: false});

                client.get(`sales/transformation-pack?start_date=${this.start_date}&end_date=${this.end_date}`).then(response => {
                    $pack_sales.rows.add(response.data);
                    $pack_sales.columns.adjust().draw();
                    $pack_sales.responsive.recalc();
                });
            },
            viewElitePackSales() {
                this.report_type = "ELITE_PACK";
                $pack_sales.clear().draw();
                $('#modal-view-pack-sales-title').text("Elite Pack Sales");
                $('#modal-view-pack-sales').modal({backdrop: 'static', keyboard: false});

                client.get(`sales/elite-pack?start_date=${this.start_date}&end_date=${this.end_date}`).then(response => {
                    $pack_sales.rows.add(response.data);
                    $pack_sales.columns.adjust().draw();
                    $pack_sales.responsive.recalc();
                });
            },
            viewFamilyElitePackSales() {
                this.report_type = "FAMILY_PACK";
                $pack_sales.clear().draw();
                $('#modal-view-pack-sales-title').text("Family Elite Pack Sales");
                $('#modal-view-pack-sales').modal({backdrop: 'static', keyboard: false});

                client.get(`sales/family-elite-pack?start_date=${this.start_date}&end_date=${this.end_date}`).then(response => {
                    $pack_sales.rows.add(response.data);
                    $pack_sales.columns.adjust().draw();
                    $pack_sales.responsive.recalc();
                });
            },
            viewEndorsers(user_id) {
                this.report_type = "TOP_ENDORSERS_ENDORSER";
                this.user_id = user_id;
                $endorsers.clear().draw();
                $endorsers.responsive.recalc();
                $('#modal-view-endorsers').modal({backdrop: 'static', keyboard: false});

                client.get(`endorsers/${user_id}/endorsers?start_date=${this.start_date}&end_date=${this.end_date}`).then(response => {
                    $endorsers.rows.add(response.data);
                    $endorsers.columns.adjust().draw();
                    $endorsers.responsive.recalc();
                });
            },
            download() {
                if(this.is_downloading) return;

                this.is_downloading = true;
                client.get(`download?report_type=${this.report_type}&start_date=${this.start_date}&end_date=${this.end_date}&user_id=${this.user_id}`).then(response => {
                    let link = response.data.link;
                    this.is_downloading = false;

                    if(!!link) {
                        window.location = link;
                    }
                });
            },
            downloadTopEndorser() {
                if(this.is_downloading_top_endorser) return;

                this.is_downloading_top_endorser = true;
                client.get(`download?report_type=TOP_ENDORSERS&start_date=${this.start_date}&end_date=${this.end_date}`).then(response => {
                    let link = response.data.link;
                    this.is_downloading_top_endorser = false;
                    window.location = link;
                });
            },
            axiosErrorHandler(error) {
                this.is_processing = 0;

                let parse = commissionEngine.parseAxiosErrorData(error.response.data)

                this.error = parse.message;

                swal(this.error, "", 'error');
            },
        }
    });


}(jQuery, window.commissionEngine.API_URL, Vue, swal, axios, window.location, moment));

File: dfodata/pages/admin_dashboard.pl
print <<EOS;

<link rel="stylesheet" href="$commission_engine_api_url/css/app.css?v=$app_css_version" />
<link rel="stylesheet" href="$commission_engine_api_url/css/datepicker.css?v=$app_css_version" />
<link rel="stylesheet" href="https://cdn.datatables.net/responsive/2.2.3/css/responsive.bootstrap.min.css">
<style>

    .text-number {
        text-align: right;
        padding-right: 5px;
    }

    .tool-container .col-md-12 {
        padding-right: 15px;
        padding-left: 15px;
    }

    .tool-container .modal-body {
        overflow-x: hidden;
    }

    #table-top-endorsers {
        margin-top: 0px !important;
    }

    .endorser-purchase-note {
        font-style: italic;
        font-size: small;
        color: #8a6d3b;
        text-align: right;
    }

    .modal th {
        text-align: center !important;
    }

    #table-new-members > thead > tr > th {
        font-size: 12px !important;
    }

</style>


<div id="dashboard" class="dashboard tool-container tool-container--default" v-cloak>
    <div class="row">
        <div class="col-md-12">
            <h4 class="admin-money-title">Dashboard</h4>
        </div>
    </div>

    <div class="row">
        <div class="col-md-12">
            <form class="form-horizontal">
                <div class="form-group">
                    <div class="form-sub-group col-sm-3 col-md-3 col-lg-3">
                        <label for="start-date">From</label>
                        <div class="input-group start-date">
                            <input type="text" id="start-date" class="form-control"> 
                            <span class="input-group-addon date-from-icon">
                                <i class="glyphicon glyphicon-calendar"></i>
                            </span>
                        </div>
                    </div>
                    <div class="form-sub-group col-sm-3 col-md-3 col-lg-3">
                        <label for="end-date">To</label>
                        <div class="input-group start-date">
                            <input type="text" id="end-date" class="form-control"> 
                            <span class="input-group-addon date-from-icon">
                                <i class="glyphicon glyphicon-calendar"></i>
                            </span>
                        </div>
                    </div>
                </div>
                
            </form>
        </div>
    </div>

    <div class="row margin-bottom">

        <div class="col-md-3">
            <button id="btn-view" type="button" class="btn btn-block btn-primary flat" v-on:click="view">View Report</button>
        </div>
        <div class="col-md-3"></div>
        <div class="col-md-3"></div>
        <div class="col-md-3"></div>
        
    </div>

    <div class="row">
        <div class="col-md-4">
            <div class="table-responsive">
                <table class="table table-bordered">
                    <thead class="table__header table__header--bg-primary">
                    <tr class="table__row">
                        <th>Description</th>
                        <th>Amount</th>
                    </tr>
                    </thead>
                    <tbody>
                    <tr class="table__row">
                        <td>New Customers</td>
                        <td class="text-number">
                            <i v-if="new_customer_count === null" class="fa fa-spinner fa-spin"></i>
                            <a class="btn-link" v-else v-on:click="viewNewCustomers">{{new_customer_count}}</a>
                        </td>
                    </tr>
                    <tr class="table__row">
                        <td>New Customers with $autoship</td>
                        <td class="text-number">
                            <i v-if="new_customer_with_product_subscription_count === null" class="fa fa-spinner fa-spin"></i>
                            <a class="btn-link" v-else v-on:click="viewNewCustomersWithProductSubscription">{{new_customer_with_product_subscription_count}}</a>
                        </td>
                    </tr>
                    <tr class="table__row">
                        <td>New $affiliate_plural</td>
                        <td class="text-number">
                            <i v-if="new_endorser_count === null" class="fa fa-spinner fa-spin"></i>
                            <a class="btn-link" v-else v-on:click="viewNewEndorsers">{{new_endorser_count}}</a>
                        </td>
                    </tr>
                    <tr class="table__row">
                        <td>New $affiliate_plural with $autoship</td>
                        <td class="text-number">
                            <i v-if="new_endorser_with_product_subscription_count === null" class="fa fa-spinner fa-spin"></i>
                            <a class="btn-link" v-else v-on:click="viewNewEndorsersWithProductSubscription">{{new_endorser_with_product_subscription_count}}</a>
                        </td>
                    </tr>
                    <tr class="table__row">
                        <td>Sample Pack 1</td>
                        <td class="text-number">
                            <i v-if="customer_transformation_pack_total_sales === null" class="fa fa-spinner fa-spin"></i>
                            <a class="btn-link" v-else v-on:click="viewCustomerTransformationPackSales">{{customer_transformation_pack_total_sales | money}}</a>
                        </td>
                    </tr>
                    <tr class="table__row">
                        <td>Sample Pack 2</td>
                        <td class="text-number">
                            <i v-if="transformation_pack_total_sales === null" class="fa fa-spinner fa-spin"></i>
                            <a class="btn-link" v-else v-on:click="viewTransformationPackSales">{{transformation_pack_total_sales | money}}</a>
                        </td>
                    </tr>
                    <tr class="table__row">
                        <td>Sample Pack 3</td>
                        <td class="text-number">
                            <i v-if="elite_pack_total_sales === null" class="fa fa-spinner fa-spin"></i>
                            <a class="btn-link" v-else v-on:click="viewElitePackSales">{{elite_pack_total_sales | money}}</a>
                        </td>
                    </tr>
                    <tr class="table__row">
                        <td>Sample Pack 4</td>
                        <td class="text-number">
                            <i v-if="family_elite_pack_total_sales === null" class="fa fa-spinner fa-spin"></i>
                            <a class="btn-link" v-else v-on:click="viewFamilyElitePackSales">{{family_elite_pack_total_sales | money}}</a>
                        </td>
                    </tr>
                    <tr class="table__row">
                        <td>Average Reorder</td>
                        <td class="text-number">
                            <i v-if="average_reorder === null" class="fa fa-spinner fa-spin"></i>
                            <span v-else>{{average_reorder}}%</span>
                        </td>
                    </tr>
                    <tr class="table__row">
                        <td>Viral Index <span v-if="viral_index_start_date !== null">({{viral_index_start_date}} - {{viral_index_end_date}})</span></td>
                        <td class="text-number">
                            <i v-if="viral_index === null" class="fa fa-spinner fa-spin"></i>
                            <span v-else>{{viral_index}}</span>
                        </td>
                    </tr>
                    </tbody>
                </table>
            </div>
        </div>
        <div class="col-md-8">
            <table id="table-top-endorsers" class="table table-striped table-bordered dt-responsive nowrap" style="width:100%">
                <thead class="table__header table__header--bg-primary">
                <tr class="table__row">
                    <th>ID</th>
                    <th>Top</th>
                    <th>$affiliate_plural</th>
                    <th>Volume</th>
                </tr>
                </thead>
            </table>


            <button type="button" v-if="top_endorser_count > 0" v-on:click="downloadTopEndorser" :disabled="is_downloading_top_endorser" class="btn btn-orange" style="margin-bottom: 10px;">
                <span v-if="is_downloading_top_endorser">Generating&hellip;</span>
                <span v-else>Download Top $affiliate</span>
            </button>
        </div>
    </div>

    <div class="modal fade" id="modal-view-new-members" role="dialog" aria-labelledby="modal-view-new-members-title">
        <div class="modal-dialog modal-lg" role="document">
            <div class="modal-content">
                <div class="modal-header">
                    <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
                    <h4 class="modal-title" id="modal-view-new-members-title">
                        New Members
                    </h4>
                </div>
                <div class="modal-body">

                    <button type="button" v-on:click="download" :disabled="is_downloading" class="btn btn-primary" style="margin-bottom: 10px;">
                        <span v-if="is_downloading">Generating&hellip;</span>
                        <span v-else>Download</span>
                    </button>

                    <table id="table-new-members" class="table table-striped table-bordered dt-responsive nowrap" style="width:100%">
                        <thead class="table__header table__header--bg-primary">
                        <tr class="table__row">
                            <th>ID</th>
                            <th class="table-new-member-type">Member</th>
                            <th>Sponsor ID</th>
                            <th>Sponsor</th>
                            <th>Sponsor Type</th>
                            <th>Order #</th>
                            <th>CV</th>
                            <th>Total \$ paid</th>
                            <th>Gift Card used</th>
                            <th>Coupon used</th>
                            <th>Shipping City</th>
                            <th>Shipping State</th>
                            <th>Product Subscription</th>
                            <th>Phone Number</th>
                        </tr>
                        </thead>
                    </table>
                </div>
                <div class="modal-footer">
                    <a type="button" class="btn btn-primary" data-dismiss="modal">Close</a>
                </div>
            </div>
        </div>
    </div>

    <div class="modal fade" id="modal-view-pack-sales" role="dialog" aria-labelledby="modal-view-pack-sales-title">
        <div class="modal-dialog modal-lg" role="document">
            <div class="modal-content">
                <div class="modal-header">
                    <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
                    <h4 class="modal-title" id="modal-view-pack-sales-title">
                        Pack Sales
                    </h4>
                </div>
                <div class="modal-body">

                    <button type="button" v-on:click="download" :disabled="is_downloading" class="btn btn-primary" style="margin-bottom: 10px;">
                        <span v-if="is_downloading">Generating&hellip;</span>
                        <span v-else>Download</span>
                    </button>

                    <table id="table-pack-sales" class="table table-striped table-bordered dt-responsive nowrap" style="width:100%">
                        <thead class="table__header table__header--bg-primary">
                        <tr class="table__row">
                            <th>Purchaser ID</th>
                            <th>Purchaser</th>
                            <th>Invoice</th>
                            <th>Sponsor ID</th>
                            <th>Sponsor</th>
                            <th>Order #</th>
                            <th>CV</th>
                            <th>Total \$ paid</th>
                            <th>Gift Card used</th>
                            <th>Coupon used</th>
                            <th>Shipping City</th>
                            <th>Shipping State</th>
                            <th>Sponsor Type</th>
                        </tr>
                        </thead>
                    </table>
                </div>
                <div class="modal-footer">
                    <a type="button" class="btn btn-primary" data-dismiss="modal">Close</a>
                </div>
            </div>
        </div>
    </div>

    <div class="modal fade" id="modal-view-endorsers" role="dialog" aria-labelledby=modal-view-endorsers-title">
        <div class="modal-dialog modal-lg" role="document">
            <div class="modal-content">
                <div class="modal-header">
                    <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
                    <h4 class="modal-title" id="modal-view-endorsers-title">
                        $affiliate_plural
                    </h4>
                </div>
                <div class="modal-body">

                    <button type="button" v-on:click="download" :disabled="is_downloading" class="btn btn-primary" style="margin-bottom: 10px;">
                        <span v-if="is_downloading">Generating&hellip;</span>
                        <span v-else>Download</span>
                    </button>

                    <table id="table-modal-view-endorsers" class="table table-striped table-bordered dt-responsive nowrap" style="width:100%">
                        <thead class="table__header table__header--bg-primary">
                        <tr class="table__row">
                            <th>$affiliate ID</th>
                            <th>$affiliate</th>
                            <th>Invoice</th>
                            <th>Description</th>
                            <th>Transaction Date</th>
                            <th>Sponsor ID</th>
                            <th>Sponsor</th>
                            <th>Order #</th>
                            <th>CV</th>
                            <th>Total \$ paid</th>
                            <th>Gift Card used</th>
                            <th>Coupon used</th>
                            <th>Shipping City</th>
                            <th>Shipping State</th>
                            <th>Sponsor Type</th>
                            <th>Phone Number</th>
                        </tr>
                        </thead>
                    </table>
                    <p class="endorser-purchase-note">The invoice shown per member is the first purchased of the member as $affiliate type</p>
                </div>
                <div class="modal-footer">
                    <a type="button" class="btn btn-default" data-dismiss="modal">Close</a>
                </div>
            </div>
        </div>
    </div>

</div>

<script src="$commission_engine_api_url/js/app.js?v=$app_js_version"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/select2/4.0.7/js/select2.full.min.js"></script>
<script src="https://cdn.datatables.net/responsive/2.2.3/js/dataTables.responsive.min.js"></script>
<script src="https://cdn.datatables.net/responsive/2.2.3/js/responsive.bootstrap.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-datepicker/1.8.0/js/bootstrap-datepicker.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/moment.js/2.24.0/moment.min.js" integrity="sha256-4iQZ6BVL4qNKlQ27TExEhBN1HFPvAvAMbFavKKosSWQ=" crossorigin="anonymous"></script>
<script src="$commission_engine_api_url/js/admin_dashboard.js?v=1.0&app=$app_js_version"></script>

EOS
1;